### ISSUE TITLE
AutoSuggest component

### ISSUE THREAD

**Issue by smith-kyle** [JedWatson/react-select#1552]
It would be helpful if there was a component that wrapped `Select`, similar to `Select.Creatable`, but that's geared towards suggestions:
  - After an option is selected or arrowed over, the value populates the input and the cursor is placed at the end of the value
  - "Create option ..." is not listed when no match is found. Instead the value of the input is used as-is
  - When a custom value (not found in the options) is typed, the value is not added to the list of options

Basically, the default behavior of this component would be one-to-one with the Google search input.

Related: #1266

**Comment by bladey** [comment]
Hello -

In an effort to sustain the `react-select` project going forward, we're closing old issues / pull requests.

We understand this might be inconvenient but in the best interest of supporting the broader community we have to direct our limited efforts to maintain the latest version.

If you feel this issue / pull request is still relevant and you'd like us to review it, please leave a comment and we'll do our best to get back to you.
